azurerm/resource_arm_storage_management_policy.go

```diff
@@ -0,0 +1,389 @@
+												"tier_to_cool_after_days_since_modification_greater_than": {
+													Type:     schema.TypeInt,
+													Optional: true,
+												},
+												"tier_to_archive_after_days_since_modification_greater_than": {
+													Type:     schema.TypeInt,
+													Optional: true,
+												},
+												"delete_after_days_since_modification_greater_than": {
+													Type:     schema.TypeInt,
+													Optional: true,
+												},
+											},
+										},
+									},
+									"snapshot": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"delete_after_days_since_creation_greater_than": {
+													Type:     schema.TypeInt,
+													Optional: true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceArmStorageManagementPolicyCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Storage.ManagementPoliciesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	storageAccountId := d.Get("storage_account_id").(string)
+
+	rid, err := parseAzureResourceID(storageAccountId)
+	if err != nil {
+		return err
+	}
+	resourceGroupName := rid.ResourceGroup
+	storageAccountName := rid.Path["storageAccounts"]
+
+	name := "default" // The name of the Storage Account Management Policy. It should always be 'default' (from https://docs.microsoft.com/en-us/rest/api/storagerp/managementpolicies/createorupdate)
+
+	parameters := storage.ManagementPolicy{
+		Name: &name,
+	}
+
+	rules := d.Get("rule").([]interface{})
+	armRules, err := expandStorageManagementPolicyRules(rules)
+	if err != nil {
+		return fmt.Errorf("Error expanding Azure Storage Management Policy Rules %q: %+v", storageAccountId, err)
+	}
+
+	parameters.ManagementPolicyProperties = &storage.ManagementPolicyProperties{
+		Policy: &storage.ManagementPolicySchema{
+			Rules: armRules,
+		},
+	}
+
+	result, err := client.CreateOrUpdate(ctx, resourceGroupName, storageAccountName, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating Azure Storage Management Policy %q: %+v", storageAccountId, err)
+	}
+
+	d.SetId(*result.ID)
+
+	return resourceArmStorageManagementPolicyRead(d, meta)
+}
+
+func resourceArmStorageManagementPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Storage.ManagementPoliciesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	storageAccountId := d.Get("storage_account_id").(string)
+
+	rid, err := parseAzureResourceID(storageAccountId)
+	if err != nil {
+		return err
+	}
+	resourceGroupName := rid.ResourceGroup
+	storageAccountName := rid.Path["storageAccounts"]
+
+	result, err := client.Get(ctx, resourceGroupName, storageAccountName)
+	if err != nil {
+		return err
+	}
+	d.SetId(*result.ID)
+
+	if result.Policy != nil {
```

Comment:
We can rewrite this to be a little cleaner
```suggestion
	if policy := result.Policy; policy != nil {
```